### ISSUE TITLE
Path not set on `schemaIssue()`

### ISSUE THREAD

**Issue by richardvanbergen** [fabian-hiller/valibot#303]
I ran into an issue on [react-hook-forms valibot resolver](https://github.com/react-hook-form/resolvers/tree/master/valibot). The bug itself is a result of the resolver but I think there's maybe a small design flaw in valibot that is the root cause so I wanted to ask about it here.

The issue is that the [schemaIssue() does not return a path](https://github.com/fabian-hiller/valibot/blob/671ff7db64fb8f04209014f476855b56fe682045/library/src/schemas/variant/variant.ts#L145). As a result, there's an exit condition [that's not being addressed in the resolver](https://github.com/react-hook-form/resolvers/blob/3805d3eeae950ba6540f5d79d264c49287008d6a/valibot/src/valibot.ts#L42). So there's two sides to this bug and I think there's a case for addressing it here.

Take a look at this simplified schema:

```
const numberSchema = object({
  type: literal("number")
})

const stringSchema = object({
  type: literal("string")
})

const typeSchema = variant("type", [numberSchema, stringSchema])

parse(typeSchema, {
  type: undefined
})
```

For most cases, this isn't an issue. We'll throw an error saying that because the type isn't set we can't select a variant to validate with. 

For form validation libraries though it's likely that type isn't set and that we're trying to validate it. It's expected behaviour that type could be undefined, it's also expected that the this should fail with a path set so we can map it back to the field that hasn't set it properly.

Because variant isn't a `validation` but a `schema` by design, the validation path isn't set and the only options the users have are:

* Have a default value always
* Fix the issue in react-hook-form (which I will also file a bug for) but not map the error back to the field.
* Some other workaround on the users side. Maybe some sort of pre-processing?

Zod does address this as validation error through `discriminatedUnion` and I think that's the correct approach.

**Comment by fabian-hiller** [comment]
Thank you for creating this issue. `variant` is the same as `discriminatedUnion` in Zod. I think we can solve this problem for Valibot by adding a path pointing to the discriminator key in the issue object. That way the resolver will automatically add the error message to the right field.

**Comment by richardvanbergen** [comment]
Hey @fabian-hiller thanks for the fast response. I've submitted a PR for the resolver above. I would help here too but I don't know enough about the repos structure and how you might want to address it but let me know if there's anything else I can do.

**Comment by fabian-hiller** [comment]
Thank you! I will investigate and fix it next week.

**Comment by fabian-hiller** [comment]
This is fixed in v0.25.0

**Comment by richardvanbergen** [comment]
Congrats and thanks!
